shark-android/src/test/java/shark/AndroidReferenceMatcher_XIAMI__RESOURCES_IMPL_Test.kt

```diff
@@ -0,0 +1,173 @@
+package shark
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import java.io.File
+import shark.AndroidReferenceMatchers.Companion.HMD_GLOBAL
+import shark.AndroidReferenceMatchers.Companion.INFINIX
+import shark.AndroidReferenceMatchers.Companion.LENOVO
+import shark.AndroidReferenceMatchers.Companion.NVIDIA
+import shark.AndroidReferenceMatchers.Companion.XIAOMI
+import shark.HprofHeapGraph.Companion.openHeapGraph
+import shark.ReferencePattern.StaticFieldPattern
+import shark.ValueHolder.IntHolder
+
+/**
+ * Tests for [AndroidReferenceMatchers.XIAMI__RESOURCES_IMPL] known leaks, to ensure we
+ * don't report the ones we know about as application leaks, and do report the ones we don't.
+ */
+class AndroidReferenceMatcher_XIAMI__RESOURCES_IMPL_Test {
+
+  /**
+   * It's known that some device manufacturers on SDK 30 and above leak
+   * a statically held [android.content.Context] from a class [android.content.res.ResourcesImpl]
+   * in a variable named [mContext].
+   */
+  private val expectedKnownClassLeakingContext = "android.content.res.ResourcesImpl"
+  private val expectedKnownReferenceName = "mAppContext"
+
+  /**
+   * Arbitrarily stops at 60, but will we really be making Android apps by then?
+   * #AI-will-take-over
+   */
+  @Test fun `Expect anything under SDK 30 unknown`() {
+    for(i in 25..60) {
```

Comment:
Going a bit overboard with 60 aren't we :) ?

Reply:
I'm setting up for the Y3k switch